packages/parser/src/dast-normalize/normalize-dast.ts

```diff
@@ -46,6 +53,29 @@ const pluginChangeCdataToText: Plugin<[], DastRoot, DastRoot> = () => {
     };
 };
 
+/**
+ * Interpret all text nodes that are children of the root node as Markdown, transforming them to HTML.
+ */
+const pluginTextToMarkdown: Plugin<[], DastRoot, DastRoot> = () => {
+    return (tree) => {
+        visit(tree, (node, visitInfo) => {
+            // All text nodes that are children of the root node are interpreted as Markdown.
+            if (node.type === "text" && visitInfo.parents.length === 1) {
+                const html = commonMarkRenderer.render(
+                    commonMarkParser.parse(node.value),
+                );
```

Comment:
Very interesting strategy. It's certainly quick for now, but I think it would be better to convert the AST directly so we have full control over the nodes created (e.g, in code blocks, etc.)

Reply:
Do you mean that it's possible we'll be using Markdown not just with text nodes, but in other places (inside code blocks, for example)? If so, then your comment below makes a lot of sense -- define where Markdown belongs and where it doesn't.